File: Practica 1/Punto.py
import math

class Punto:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def coord_cartesianas(self):
        return (self.x, self.y)

    def coord_polares(self):
        r = math.sqrt(self.x**2 + self.y**2)
        theta = math.atan2(self.y, self.x)
        return (r, theta)

    def __str__(self):
        return f"Punto(x={self.x}, y={self.y})"

# Ejemplo de uso
p = Punto(3, 4)
print(p.coord_cartesianas())  # Salida: (3, 4)
print(p.coord_polares())      # Salida: (5.0, 0.9272952180016122)
print(p)                      # Salida: Punto(x=3, y=4)

File: Practica 1/PuntoenJava/src/punto/Punto.java
package punto;

public class Punto {
	public double x;
    public double y;

    public Punto(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public String coord_cartesianas() {
        return "(" + x + ", " + y + ")";
    }

    public String coord_polares() {
        double r = Math.sqrt(x * x + y * y);
        double theta = Math.atan2(y, x);
        return "(" + r + ", " + theta + ")";
    }

    @Override
    public String toString() {
        return "Punto(x=" + x + ", y=" + y + ")";
    }

    // Ejemplo de uso
    public static void main(String[] args) {
        Punto p = new Punto(3, 4);
        System.out.println(p.coord_cartesianas());  // Salida: (3.0, 4.0)
        System.out.println(p.coord_polares());      // Salida: (5.0, 0.9272952180016122)
        System.out.println(p);                     // Salida: Punto(x=3.0, y=4.0)
    }
}
